libs/ui/src/diagnostics/test_utils.ts

```diff
@@ -0,0 +1,44 @@
+import { within } from '@testing-library/react';
+import { assertDefined } from '@votingworks/basics';
+import { VxScreen } from '../themes/render_with_themes';
+import { DiagnosticSectionTitle } from './types';
+
+function expectTextInSection(
+  screen: VxScreen,
+  headerText: string,
+  expectedText: string | RegExp
+) {
+  const section = screen.getByText(headerText).closest('section');
+  expect(
+    within(
+      assertDefined(
+        section,
+        `Couldn't find section with header '${headerText}'`
+      )
+    ).getByText(expectedText)
+  ).toBeDefined();
+}
+
+export function expectDetected(
```

Comment:
These helpers pertain to motivation #1 in the PR description and will be used in a future PR